### DISCUSSION TITLE
RS485 Support?

### DISCUSSION THREAD

**Discussion by DjordjeMandic** [stefanrueger/urboot#40] [Ideas]
Is there support for RS485 on USART0?

**Comment by stefanrueger** [comment]
Interesting question. I haven't thought about that. What's needed? Half-duplex? Switching between rx and tx mode in putch and getch?

**Reply by DjordjeMandic** [reply]
Yeah. MAX485 is most common. I guess that device would need to have some kind of ID on bus. IDK how to do restart. maybe enter bootloader from app code with value passed in one register to tell it to go into rs485 mode with specific id.

![image](https://github.com/user-attachments/assets/2d63eca5-1c50-427d-9cea-dc0a8b8f1448)

**Comment by stefanrueger** [comment]
I don't have test equipment for RS485 nor do I know what exactly needs doing. So, out of necessity RS485 support won't be a priority for the urboot project. If someone submits a working urboot modification for programming via an RS485 line I'd be happy to consider what's involved in merging that into the project.

It sounds like you have a specific multidrop application in mind what with bus ids and all. So, without knowing more this might be more involved than switching tx/rx for half-duplex. Also note that urboot isn't designed to be called from the application: it assumes a MCU state fresh from a reset.  Have you checked whether how other bootloaders implement RS845? Maybe looking at https://github.com/mikalai-kavalchuk/optiboot/tree/RS485_support will give you ideas how to accomplish what you need?
